FlashCap.Core/Devices/DirectShowDevice.cs

```diff
@@ -174,13 +175,24 @@ private sealed class SampleGrabberSink :
 
                     ///////////////////////////////
 
-                    var captureGraphBuilder = NativeMethods_DirectShow.CreateCaptureGraphBuilder();
+                    captureGraphBuilder = NativeMethods_DirectShow.CreateCaptureGraphBuilder();
                     if (captureGraphBuilder.SetFiltergraph(this.graphBuilder) < 0)
                     {
                         throw new ArgumentException(
                             $"FlashCap: Couldn't set graph builder: DevicePath={devicePath}");
                     }
 
+                    ///////////////////////////////
+                    
+                    captureGraphBuilder.FindInterface(Guid.Empty, Guid.Empty, captureSource, NativeMethods_DirectShow.IAMVideoProcAmpHelper.GUID, out object? videoProcAmpObject);
```

Comment:
on initialize, we can feed the properties dictionary

I've decided to start with `IAMVideoProcAmp`

I've defined a CaptureDeviceProperty which is implemented for each devices, directshow in this case, the abstraction is based on this one since it's the only one I know of. slightly opiniated.